Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -352,28 +353,17 @@ int CLuaResourceDefs::GetResourceGUIElement(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM)
+CClientEntity* CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM, std::optional<CResource*> resource)
 {
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
+    CResource* pResource = resource.value_or(&lua_getownerresource(luaVM));
```

Comment:
We are trying not to use Hungarian notation anymore, so please rename it to "resource"

Reply:
I cannot rename it to `resource` as that name is defined as a parameter

Reply:
Yes, I can rename it but not to resource I meant 😅
Will change in a sec